src/PrestaShopBundle/Translation/PrestaShopTranslatorTrait.php

```diff
@@ -58,20 +58,28 @@ public function trans($id, array $parameters = [], $domain = null, $locale = nul
             $locale = null;
         }
 
-        $translated = parent::trans($id, [], $this->normalizeDomain($domain), $locale);
+        $isSprintfString = $this->isSprintfString($id);
+        $translated = parent::trans(
+            $id,
+            $isSprintfString ? [] : $parameters,
```

Comment:
Shouldn't it be the parent responsibility to know if $parameters should be used or not ? :thinking:

Reply:
I am not sure here the parent is symfony's translator component `src/Symfony/Component/Translation/Translator.php`. And the function signature assign a default [] value to `$parameters` param. In our case we want to pass the empty array (or nothing) if `$this->isSprintfString($id);`  is true (meaning it has special characters that need to be replace with vsprintf().

Reply:
I see the idea, but I'm a bit concerned about this `isSprintfString` feature because we don't really know for which use cases it is used 🤷‍♂️

Probably a legacy use case, maybe @eternoendless or @kpodemski have an idea of use cases where this is needed in modules? This would help us figure out which specific use cases still need to be handled here, and maybe (I'm dreaming) we could add some integration tests for those specific use cases (along with the plural ones that are the initial problem of this PR).

By the way @yanmakouf I think the `strtr` at the end of this method is overkill, because `trans` already performs this, now that the parameters are correctly passed to the parent class we don't need to handle this in this override method. So only the `sprintf` particular use case should remain.

Reply:
By the way, @yanmakouf I think it would be worth adding an integration test class that checks a few specific wordings like:
- a regular Symfony wording without parameters
- a Symfony wording with parameters
- a plural wording with parameters (we can use the one from the Length validator)
- a legacy wording relying on sprintf

The idea is not to check all wordings, but at least an example of each type of wording to make sure we don't break anything in the future. And maybe one day we'll be able to clean this trait completely and remove the tests, but until then it's better to be safe.

Reply:
Thanks for your review @jolelievre, I'll try'n propose something more complete :)

Reply:
Doing this will prevent falling back to the legacy translation system when using parameters, because the translated result will include interpolated parameters, and one of the hints we use to fall back is that the translated string is the same as the string before translation (see `shouldFallbackToLegacyModuleTranslation()`).

I don't know why that hint is still used, since `shouldFallbackToLegacyModuleTranslation()` already checks if the domain starts with "Module." AND that the message is not found within the catalog. That should be enough to tell that the message should be translated with the legacy system. Or perhaps I forgot something.

Reply:
Regarding `isSprintfString`, it's used to tell if the message should be interpolated with sprintf or with strtr. The first uses placeholders like "%s" while the latter uses placeholders like "%foo%".